micro-ui/web/packages/modules/workbench/src/configs/UICustomizations.js

```
@@ -1083,49 +1084,63 @@ export const UICustomizations = {
+          const pollForGeneratedResource = async (maxAttempts = 10, intervalMs = 3000) => {
+            for (let attempt = 0; attempt < maxAttempts; attempt++) {
+              const resFile = await generateTemplate();
+              const resource = resFile?.GeneratedResource?.[0];
+              if (resource?.fileStoreid) return resource;
+              if (resource?.status === "failed") return null;
+              if (attempt < maxAttempts - 1) {
+                await new Promise((resolve) => setTimeout(resolve, intervalMs));
+              }
             }
+            return null;
           };
```

Biome (2.5.3)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)